### PR TITLE
strict pandas types

### PR DESCRIPTION
**Pull request by martacki**
Closes # (if applicable).

## Checklist

- [ ] Code changes are sufficiently documented; i.e. new functions contain docstrings and further explanations may be given in `doc`.
- [ ] Unit tests for new features were added (if applicable).
- [ ] Newly introduced dependencies are added to `environment.yaml`, `environment_docs.yaml` and `setup.py` (if applicable).
- [ ] A note for the release notes `doc/release_notes.rst` of the upcoming release is included.
- [ ] I consent to the release of this PR's code under the MIT license.

### CODE REVIEW

File: pypsa/io.py [hunk 1]
```diff
@@ -904,6 +904,8 @@ def import_components_from_dataframe(network, dataframe, cls_name):
         else:
             if static_attrs.at[k, "type"] == "string":
                 dataframe[k] = dataframe[k].replace({np.nan: ""})
+            if static_attrs.at[k, "type"] in ["float", "int"]:
```
Comment (FabianHofmann) [inline pypsa/io.py:907]:
```suggestion
            if static_attrs.at[k, "type"] == "int":
```

Comment (FabianHofmann) [reply pypsa/io.py:907]: this makes more sense or @martacki

File: pypsa/io.py [hunk 2]
```diff
@@ -904,6 +904,8 @@ def import_components_from_dataframe(network, dataframe, cls_name):
...
+                dataframe[k] = dataframe[k].replace({np.nan: 0})
```
Comment (FabianHofmann) [inline pypsa/io.py:908]:
```suggestion
                dataframe[k] = dataframe[k].fillna(0)
```

Comment (martacki) [reply pypsa/io.py]: Just tried to stick to the old convention, if we use this one then I'd suggest to use `dataframe[k] = dataframe[k].fillna("")` as well (1-2 lines earlier)

**Pull request merged**